src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -107,17 +124,27 @@ class TaskDeploymentService {
     /**
      * Helper method to initialise the lateinit vars [taskAssumeRoleString] and [taskRolePolicyString] by
      * converting the associated `@Value` parameters to Strings and replacing `ADDITIONAL_PERMISSIONS` in
-     * [taskRolePolicyString] with the provided [additionalPermissions]
+     * [taskRolePolicyString] with the provided [listOfParameters]
      *
-     * @return [Pair] of [taskRolePolicyString] to [taskAssumeRoleString] for ease of access.
+     * @return [taskRolePolicyString] and [taskAssumeRoleString].
      */
-    fun parsePolicyDocuments(additionalPermissions: List<String>): Pair<String, String> {
-        logger.info("Adding permissions to containers", "permissions" to additionalPermissions.joinToString())
-        val permissionsJson = additionalPermissions.joinToString(prefix = "\"", separator = "\",\"", postfix = "\"")
-
-        taskAssumeRoleString = taskAssumeRoleDocument.inputStream.bufferedReader().use { it.readText() }
-        taskRolePolicyString = taskRolePolicyDocument.inputStream.bufferedReader().use { it.readText() }
-                .replace("ADDITIONAL_PERMISSIONS", permissionsJson)
-        return taskRolePolicyString to taskAssumeRoleString
+    fun parsePolicyDocument(resource: Resource, placeholderAndReplacements: Map<String, List<String>>): String {
+        var resourceToString = resource.inputStream.bufferedReader().use { it.readText() }
```

Comment:
This level of complex string replacement is a huge code smell to me and introduces a lot of potential instability to the program. I think that since all of our documents are in JSON format I'd like to see them parsed into actual JSON objects which can be handled in a more idempotent way.
Doing this will also help to add context, for example showing that the docs are JSON and the keys which these values are to be associated with (at the moment we are using arbitrary replacement strings). This could be done by using Jackson's `ObjectMapper` for example